24.md

```diff
@@ -0,0 +1,497 @@
+NIP-24
+======
+
+Sealed Private Messages and Private Group Chats
+-----------------------------------------------
+
+`draft` `optional` `author:vitorpamplona` `author:kieran` `author:paulmillr`
+
+This NIP creates a metadata-minimized encrypted messaging protocol for direct messages and closed group chats with the following privacy guarantees: 
+1. Messages are encrypted to each participant's public key individually. 
+2. Chat participant identities, each message's real date and time, event kinds, and any other tags are hidden from the public.
+3. Senders and Receivers cannot be linked with public information alone, not even their random keys.
+4. There is no central queue, channel or otherwise converging event id to link or count all messages in the same group. 
+5. Minimal trust in counterparties: Counterparties cannot expose the details of your message, including the metadata, without exposing their entire profile (private key)
+6. There is no moderation role (i.e. no group admins, no invitations or bans). 
+7. Messages can be recoverable in any client with the receiver or the sender's private key
+8. Users and clients can opt-in for Disappearing Messages that are not recoverable with the private key
+9. The protocol's messages must flow through public relays. Private relays can increase privacy further, but they are not needed.
+10. The protocol becomes extensible to any other event kind to be transmitted privately by itself (private likes, private reports, private long-form-content, etc)
+
+The resulting protocol is a convergence of many past proposals, including [NIP-44](https://github.com/nostr-protocol/nips/pull/574), [NIP-59](https://github.com/nostr-protocol/nips/pull/468), [NIP-24](https://github.com/nostr-protocol/nips/pull/56), [NIP-103](https://github.com/nostr-protocol/nips/pull/499), [NIP-76](https://github.com/d-krause/nostr-nips/blob/nip76-draft-2/76.md). It brings their best ideas together is a slighly different approach. 
+
+## Overview
+
+This protocol uses two main concepts to protect the transmission of a target event: Sealed Gossip Events and Gift Wraps ([NIP-59](https://github.com/nostr-protocol/nips/pull/468)). The target event is first **unsigned** (to make it unverifiable if it leaks -> "Gossip"), then Sealed, and then GiftWrapped.
```

Comment:
Gossip doesn't seem quite right, what about "Rumor"?

Reply:
Yeah, I am struggling with these names as well. :(

Reply:
Maybe we can change from Sealed Gossip to Ratifed Rumor. It's strange to Seal a Rumor.

Reply:
I don't mind that, I like the Sealed metaphor too. A letter with no signature means nothing, but if you put it in an envelope with a wax seal it's authenticated. You wouldn't ratify a rumor either 😂

Reply:
Could go with metonymy and just call the outer event a "Seal" since that's effectively the role it's performing

Reply:
In my implementation I ended up doing "rumor/seal/wrapper"